### PR TITLE
GEODE-3308: Lucene rolling upgrade and backwards compatibility tests …

### PR DESCRIPTION
**Pull request by nabarunnag**
…added
 

Thank you for submitting a contribution to Apache Geode.

In order to streamline the review of the contribution we ask you
to ensure the following steps have been taken:

### For all changes:
- [x] Is there a JIRA ticket associated with this PR? Is it referenced in the commit message?

- [x] Has your PR been rebased against the latest commit within the target branch (typically `develop`)?

- [x] Is your initial contribution a single, squashed commit?

- [x] Does `gradlew build` run cleanly?

- [x] Have you written or updated unit tests to verify your changes?

- [ ] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](http://www.apache.org/legal/resolved.html#category-a)?

### Note:
Please ensure that once the PR is submitted, you check travis-ci for build issues and
submit an update to your PR as soon as possible. If you need help, please send an
email to dev@geode.apache.org.

### CODE REVIEW

**Comment by nabarunnag** [comment]
* These tests were ported from RollingUpgradeDUnitTest and RollingUpgrade2DUnitTest
* Instead of put and get verification, it is now lucent query result size verification

Potential Reviewers
@ladyVader @jhuynh1 @DivineEnder

File: geode-lucene/src/test/java/org/apache/geode/cache/lucene/LuceneSearchWithRollingUpgradeDUnit.java [hunk 1]
```diff
@@ -0,0 +1,1044 @@
...
+    IgnoredException.addIgnoredException(
+        "cluster configuration service not available|ConflictingPersistentDataException");
+  }
+
+  public LuceneSearchWithRollingUpgradeDUnit(String version) {
+    oldVersion = version;
+  }
+
+  @Test
+  public void fluceneQueryReturnsCorrectResultsAfterServersRollOverOnPartitionRegion()
```
Comment (ladyVader) [inline geode-lucene/src/test/java/org/apache/geode/cache/lucene/LuceneSearchWithRollingUpgradeDUnit.java]: Does the f in fluceneQueryReturns... mean anything or is that just a typo?

Comment (nabarunnag) [inline geode-lucene/src/test/java/org/apache/geode/cache/lucene/LuceneSearchWithRollingUpgradeDUnit.java]: @ladyVader , its a typo 😔😔😔
I will get it fixed

File: geode-lucene/src/test/java/org/apache/geode/cache/lucene/LuceneSearchWithRollingUpgradeDUnit.java [hunk 2]
```diff
@@ -0,0 +1,1044 @@
...
+    Object manager = getRMMethod.invoke(cache);
+
+    Method createRebalanceFactoryMethod = manager.getClass().getMethod("createRebalanceFactory");
+    createRebalanceFactoryMethod.setAccessible(true);
+    Object rebalanceFactory = createRebalanceFactoryMethod.invoke(manager);
+    Method m = rebalanceFactory.getClass().getMethod("start");
+    m.setAccessible(true);
+    Object op = m.invoke(rebalanceFactory);
+
+    // Wait until the rebalance is completex
```
Comment (jhuynh1) [inline geode-lucene/src/test/java/org/apache/geode/cache/lucene/LuceneSearchWithRollingUpgradeDUnit.java]: typo

File: geode-lucene/src/test/java/org/apache/geode/cache/lucene/LuceneSearchWithRollingUpgradeDUnit.java [hunk 3]
```diff
@@ -0,0 +1,1044 @@
...
+    } catch (Exception e) {
+      Thread.currentThread().interrupt();
+      throw e;
+    }
+  }
+
+  /**
+   * Starts a locator with given configuration.
+   *
+   * @param props TODO
```
Comment (jhuynh1) [inline geode-lucene/src/test/java/org/apache/geode/cache/lucene/LuceneSearchWithRollingUpgradeDUnit.java]: remove todo?

**Pull request closed without merging**